desc/geometry/curve.py

```diff
@@ -225,6 +225,66 @@ def from_input_file(cls, path):
         )
         return curve
 
+    @classmethod
+    def from_values(cls, coords, N=10, NFP=1, basis="rpz", name="", sym=False):
+        """Fit coordinates to FourierRZCurve representation.
+
+        Parameters
+        ----------
+        coords: ndarray, shape (num_coords,3)
+            coordinates to fit a FourierRZCurve object with each column
+            corresponding to xyz or rpz depending on the basis argument.
+        N : int
+            Fourier resolution of the new R,Z representation.
+        NFP : int
+            Number of field periods, the curve will have a discrete toroidal symmetry
+            according to NFP.
+        basis : {"rpz", "xyz"}
+            basis for input coordinates. Defaults to "rpz"
+
+        Returns
+        -------
+        curve : FourierRZCurve
+            New representation of the curve parameterized by Fourier series for R,Z.
+
+        """
+        if basis == "rpz":
+            coords_rpz = coords
+            coords_xyz = rpz2xyz(coords)
+        else:
+            coords_rpz = xyz2rpz(coords)
+            coords_xyz = coords
+        R = coords_rpz[:, 0]
+        phi = coords_rpz[:, 1]
+
+        X = coords_xyz[:, 0]
+        Y = coords_xyz[:, 1]
+        Z = coords_rpz[:, 2]
+
+        # easiest to check closure in XYZ coordinates
+        X, Y, Z, _, _, _, input_curve_was_closed = _unclose_curve(X, Y, Z)
+        if input_curve_was_closed:
+            R = R[0:-1]
+            phi = phi[0:-1]
+            Z = coords_rpz[0:-1, 2]
+        # check if any phi are negative, and make them positive instead
+        # so we can more easily check if it is monotonic
+        inds_negative = np.where(phi < 0)
+        phi = phi.at[inds_negative].set(phi[inds_negative] + 2 * np.pi)
```

Comment:
Yes, because the `atan2` function in `xyz2rpz` does not give values from `0` to `2pi` but rather from `-pi` to `pi`, so if you convert an array of x,y values that corresponds to `0` to `2pi` in phi, the `xyz2rpz` function will give `phi` values from `0` to `pi` then from `-pi` back to `0`, which makes checking monotonicity hard